src/bussinessLogic/ManageArray.java

```diff
@@ -65,31 +64,22 @@ public void shuffleCharArray(char[] charArray) {
         }
     }
 
-    /**
-     * transfer the sorted chars to a char array
-     */
-    public void setCharArrayFromSortedMap() {
-        charArrayFromSortedMap = manageMap.getSortedMap().keySet().stream().map(String::valueOf).collect(Collectors.joining())
-                .toCharArray();
-
-    }
-
     /**
      * generate an array with characters repeating proportionally to the reaction
      * time for each char
      */
     public void setPracticeArray() {
-
+        char[] rankingArray = manageMap.getRankingArray();
         // calculate the size of the practiceArray
         int sizeOfPracticeArray = Constants.NUMBER_OF_LETTERS * (Constants.NUMBER_OF_LETTERS + 1) / 2;
         // create the array
         practiceArray = new char[sizeOfPracticeArray];
         int index = 0;
 
         //add the chars to the array depending on the reaction time (more chars for slower RT)
-        for (int i = 0; i < charArrayFromSortedMap.length; i++) {
+        for (int i = 0; i < rankingArray.length; i++) {
             for (int j = 0; j <= i; j++) {
-                practiceArray[index] = charArrayFromSortedMap[i];
+                practiceArray[index] = rankingArray[i];
```

Comment:
Try this `for` loop like this: `practiceArray[index++] = rankingArray[i];`.
The difference is that the variable is incremented inline.

One question here: why is the value of `j not used inside the loop?

Reply:
> Try this for loop like this: practiceArray[index++] = rankingArray[i];

👍 Thanks!

> One question here: why is the value of `j not used inside the loop?

_j_ is used to set the number of times a letter is added to the array. For example the first letter is added once, the second letter twice and so on. I could as well use a while loop here, but I'll still need an index which I can corelate with the value of _i_

Reply:
As a practice, this type of explanations have their place inside the code. In this situation, please rename _j_ with _numberOfTimesALetterIsAdded_. In my opinion it makes more sense for the person who is reading the code.